extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -160,7 +177,9 @@ export class AmpSidebar extends AMP.BaseElement {
         }, ANIMATION_TIMEOUT);
       });
     });
-    this.getHistory_().push(this.close_.bind(this)).then(historyId => {
+    this.getHistory_().push(() => {
+      this.close_();
```

Comment:
So without the doing the `delay` when pushing the new state in `document-submit` the sequence goes like this:

* Open Sidebar
* Click Section 1 Link
  * We Close Sidebar
  * Which also Pops History at 1
  * And then handle the actual click on the link and Push history to 1
  * And then Popping actually happen and history is popped to 0 removing the new push with it.

I am not entirely sure yet why does this happen. But this might be dependent on when the viewer is updating their stack index and communicate the change back with the amp doc. In `viewer.html` for example we do that inside the `popstate` event handler. This is were my theory about the event queue delaying the stack index update from happening right away after the close-pop.

 (I know the `viewer.html` implementation is incomplete but my guess is the Google.com viewer is doing something similar, I've also tested with the sandbox viewer and seen the same experience)

So to summarize this is what is being logged in sequence when the timeout is not in place.
```
[VIEWER] push history to  1
[VIEWER] pop history at  1
[VIEWER] push history to  1
[VIEWER] history popped to  0 # Notice our history push is gone now!
```

After adding the timeout this gives time for `popstate` handlers to update the stack indexes. These are the logs after timeout:
```
[VIEWER] push history to  1
[VIEWER] pop history at  1
[VIEWER] history popped to  0
[VIEWER] push history to  1 # Yay pushed state preserved!
```

Does this make any sense 😄 because I am not 80% share!